packages/stems/src/components/Modal/Modal.tsx

```diff
@@ -193,13 +194,24 @@ export const Modal = forwardRef<HTMLDivElement, ModalProps>(function Modal(
 
   const transition = useTransition(isOpen, null, {
     from: { transform: 'scale(0)', opacity: 0 },
-    enter: { transform: 'scale(1)', opacity: 1 },
+    // @ts-ignore function is a valid value for enter, but the types don't acknowledge that
+    enter:
+      (item: boolean) =>
```

Comment:
Couldn't tell without looking at docs what this was - in our case, this is `isOpen` and this works because we `setIsDoneOpening(true)` with a `setImmediate` so it executes next in the event stack?

Does our version of `react-spring` have `onRest`? I think that might make more sense: https://www.react-spring.dev/docs/advanced/events#onrest

Reply:
on rest is only for sprint not transition :/ 
we should change this to spring, but i didn't want to chew that off now.

but your analysis is correct!

Reply:
Huh, the docs for the new version say it's both! Must not be supported in our version